sdks/python/apache_beam/transforms/stats.py

```diff
@@ -234,3 +239,419 @@ def extract_output(accumulator):
+                                     self._offset(new_weight))
+    return _QuantileBuffer(new_elements, new_level, new_weight)
+
+  def _collapse_if_needed(self):
+    while len(self._qs.buffers) > self._num_buffers:
+      toCollapse = []
+      toCollapse.append(heapq.heappop(self._qs.buffers))
+      toCollapse.append(heapq.heappop(self._qs.buffers))
+      minLevel = toCollapse[1].level
+
+      while len(self._qs.buffers) > 0 and self._qs.buffers[0].level == minLevel:
+        toCollapse.append(heapq.heappop(self._qs.buffers))
+
+      heapq.heappush(self._qs.buffers, self._collapse(toCollapse))
+
+  def _interpolate(self, i_buffers, count, step, offset):
+    """
+    Emulates taking the ordered union of all elements in buffers, repeated
+    according to their weight, and picking out the (k * step + offset)-th
+    elements of this list for `0 <= k < count`.
+    """
+
+    iterators = []
+    new_elements = []
+    compare_key = lambda x: self._key(x['value'])
+    for buffer_elem in i_buffers:
+      iterators.append(buffer_elem.sized_iterator())
+
+    # Python 3 `heapq.merge` support key comparison and returns an iterator and
+    # does not pull the data into memory all at once. Python 2 does not
+    # support comparison on its `heapq.merge` api, so we use the itertools
+    # which takes the `key` function for comparison and creates an iterator
+    # from it.
+    if sys.version_info[0] < 3:
+      sorted_elem = iter(
+          sorted(itertools.chain.from_iterable(iterators), key=compare_key))
+    else:
+      sorted_elem = heapq.merge(*iterators, key=compare_key)
+
+    weighted_element = next(sorted_elem)
+    current = weighted_element['weight']
+    j = 0
+    while j < count:
+      target = j * step + offset
+      j = j + 1
+      try:
+        while current <= target:
+          weighted_element = next(sorted_elem)
+          current = current + weighted_element['weight']
+      except StopIteration:
+        pass
+      new_elements.append(weighted_element['value'])
+    return new_elements
+
+  def create_accumulator(self):
+    self._qs = _QuantileState(buffer_size=self._buffer_size,
+                              num_buffers=self._num_buffers,
+                              unbuffered_elements=[], buffers=[])
+    return self._qs
+
+  def add_input(self, quantile_state, element):
+    """
+    Add a new element to the collection being summarized by quantile state.
+    """
+    if quantile_state.is_empty():
+      quantile_state.min_val = quantile_state.max_val = element
+    elif self._comparator(element, quantile_state.min_val) < 0:
+      quantile_state.min_val = element
+    elif self._comparator(element, quantile_state.max_val) > 0:
+      quantile_state.max_val = element
+    self._add_unbuffered(elem=element)
+    return self._qs
+
+  def merge_accumulators(self, accumulators):
+    """Merges all the accumulators (quantile state) as one."""
+    self._qs = self.create_accumulator()
```

Comment:
merge_accumulators is only allowed to mutate `accumulators[0]`. It should not mutate itself unless that is the case (See: https://github.com/apache/beam/blob/master/sdks/python/apache_beam/transforms/core.py#L736)

Safest option would be to change `self._qs` to `qs`.